### ISSUE TITLE
attention visualization code for fig 5

### ISSUE THREAD

**Issue by luchaoqi** [gen-omnimatte/gen-omnimatte-public#1]
Hello, thanks for open-sourcing the project!

Could you provide more details about Figure 5? If I understand correctly, the latents of the mask and the original input video are concatenated in CogVideoX’s pipeline and then patchified through CogVideoXPatchEmbed. How do you obtain the query and key for the mask and input separately?

Did you zero out each video separately, or am I perhaps missing something in the SDEdit process? Any insights would be greatly appreciated — thank you!

**Comment by yaochih** [comment]
Thank you for the detailed reading of our paper.

The attention visualization shown in Fig. 5 was conducted using the Lumiere-based model, rather than the public-release models. We currently do not plan to reimplement the attention visualization for the public models. However, we believe similar attention signals may still emerge in other video models. For instance, a recent arXiv work, [OmnimatteZero](https://arxiv.org/pdf/2503.18033), demonstrates a similar approach for extracting attention information from a DiT-based video model.

For Fig. 5, we used the pretrained *text-to-video* Lumiere model rather than our final object-effect-removal model, Casper. Our intention was to explore whether the pretrained model already had the knowledge of object-effect association.

For more details about the SDEdit process, we first added noise to an RGB video at a chosen diffusion timestep (e.g., t=0.5), then re-denoised it back to a clean video. During the denoising phase, we extracted and saved the attention maps to visualize effect associations. To obtain the response from the attention using Eq (1), the object masks were downsampled bilinearly to match the spatial resolution of the attention query and key features. We did not zero out to the video.

Hope this information helps.

**Comment by luchaoqi** [comment]
Thank you @yaochih, that's pretty helpful!

**Comment by luchaoqi** [comment]
@yaochih one quick question regarding the training time/configuration for training Casper - I didn't find relevant information in the paper. Is the configuration [here](https://github.com/gen-omnimatte/gen-omnimatte-public/blob/main/scripts/cogvideox_fun/train_casper.sh) you used to train the model? If so, what gpu setup did you use and how long would that take to train the model? Thanks!

**Comment by yaochih** [comment]
Yes, you can follow the scripts to train Cogvideox-based and Wan-based models, which are not included in our original paper. We used 4 H100-80GB GPUs to fine-tune for ~10-12K iterations. It took ~2 days, if I remember correctly.
